pkg/storage/stores/indexshipper/compactor/deletion/delete_requests_store.go

```diff
@@ -133,10 +133,11 @@ func (ds *deleteRequestsStore) writeDeleteRequest(req DeleteRequest, writeBatch
 	writeBatch.Add(DeleteRequestsTableName, fmt.Sprintf("%s:%s", cacheGenNum, req.UserID), []byte{}, generateCacheGenNumber())
 }
 
-func deleteRequestHash(userID, requestID string, sequenceNumber int64) string {
-	// Any requests made before the addition of sequence numbers won't have one
-	// Ensure backward compatibility by treating the 0th sequence number as the
-	// old format
+// backwardCompatibleDeleteRequestHash generates the hash key for a delete request.
+// Sequence numbers were added after deletion was in production so any requests made
+// before then won't have one. Ensure backward compatibility by treating the 0th
+// sequence number as the old format without any number
```

Comment:
Let us also add a note saying the sequence number of the first shard(i.e seq no 0) in the new requests would also be empty.